### PR TITLE
add Logger.Write method to interface

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- add `Write(*LogEntry)` to the logger interface
## Why
We need a public method that takes a pre-existing LogEntry to do subprocess re-logging in agent. (i.e. to consume, parse, and re-upload log lines without double-prefixing)

### CODE REVIEW

**Review by dgottlieb — COMMENTED** [review]
I think I'm being looped in a bit late here. Does this have to do with the pattern matching log stuff the agent does?

What functionality is this offering? I would expect user-code that needed access to the structure log object could get that by implementing an [appender](https://github.com/viamrobotics/rdk/blob/main/logging/appender.go#L18).

**Comment by abe-winter** [comment]
my understanding is that the current public interface will always format the input

problem we are solving: in agent, we are capturing logs from subprocesses, parsing them, and then re-logging them. in theory this is necessary to use netappender without having a 'double prefix' issue

in the current agent PR, there's a global netappender which can receive a LogEntry; this works in theory, but in the spot where I consume the global netappender, I also have access to the logger which contains that netappender. james pointed out that if I could Write(LogEntry) on the logger, we are free + easy

**Comment by Otterverse** [comment]
> I think I'm being looped in a bit late here. Does this have to do with the pattern matching log stuff the agent does?
> 
> What functionality is this offering? I would expect user-code that needed access to the structure log object could get that by implementing an [appender](https://github.com/viamrobotics/rdk/blob/main/logging/appender.go#L18).

To clarify, we have structured input already coming in. We want to write it directly, rather than generating new log entries (with potentially different data, like timestamps and filename:lineno combinations.) We want that to be written to the normal logger with the net appender.

In normal "zap" land, this would be done with something like `logger.Core().Write()`, but as this logging system in RDK was implemented as a wrapper on top of the zap logger, instead of something like a customized zap.Core, so doing that bypasses the appender functionality. This seems to be the next best solution instead, since we're not following the zap paradigm.

**Review by Otterverse — APPROVED** [review]
LGTM, but should wait until Dan is comfortable too.

File: logging/logger.go [hunk 1]
```diff
@@ -16,6 +18,7 @@ type Logger interface {
 	Sublogger(subname string) Logger
 	AddAppender(appender Appender)
 	AsZap() *zap.SugaredLogger
+	Write(*LogEntry)
```
Comment (dgottlieb) [inline logging/logger.go:21]: No idea why the linter doesn't complain about any of these. Can we explicitly document this with:
```
Unconditionally logs a LogEntry object. Specifically any configured log level is ignored.
```

I might follow up with adding a `Log(*LogEntry)` that _conditionally_ logs. So we can replace this [switch statement](https://github.com/viamrobotics/rdk/blob/927408d147a103384fd339c37df7ff5d3fa8eea0/robot/server/server.go#L435-L448) usage.

**Review by dgottlieb — CHANGES_REQUESTED** [review]

**Review by dgottlieb — APPROVED** [review]

**Pull request merged**